pkg/controllers/dynakube/otelc/certificates/certificates.go

```diff
@@ -0,0 +1,31 @@
+package certificates
+
+import "github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+
+func IsAGCertificateNeeded(dk *dynakube.DynaKube) bool {
+	if isInClusterActiveGate(dk) {
```

Comment:
```suggestion
	if isInClusterActiveGate(dk) && k.ActiveGate().HasCaCert() {
	   return true
	}
	return false
```
no need to use extra if IMO.

Reply:
why not use otelc_props for this purpose, no need to create new package and pass dk instance.

here -> https://github.com/Dynatrace/dynatrace-operator/blob/main/pkg/api/v1beta3/dynakube/otelc_props.go

Reply:
agree but I would like to emphasize that 

IsAG depends on `isInClusterActiveGate(dk)`
IsCA depends on `! isInClusterActiveGate(dk)`

because of that it never happans that both returns TRUE

Reply:
i don't get you here,

Reply:
if a {
   if b {
   }
}
same as
if a && b {}

Creating a certificate package, in my opinion, is too much. It isn't very clear. It's just what we call few "_props". You don't do anything with certificates or so.